### PR TITLE
QGA-3-Question-Model

### PR DESCRIPTION
**Pull request by ghost**
corrected questionSchema

### CODE REVIEW

File: .env.template [hunk 1]
```diff
@@ -1,11 +0,0 @@
-MONGODB_URI='mongodb://localhost:27017/quiz-game'
```
Comment (Gruseckis) [inline .env.template]: Please restore this file

File: controllers/questionController.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+   getQuestionsFromIdArray,
+   updateQuestionByID,
+   deleteQuestionByID
+} from '../models/questionModel';
+
+import AppError from '../errors/AppError';
+
+
+
+exports.addNewQuestion = async(req, res, next) => {
```
Comment (Gruseckis) [inline controllers/questionController.js]: Please create const here and export all modules at the bottom of the file

File: controllers/questionController.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+      const model = {
+         question: req.body.question,
+         correct: req.body.correct,
+         answers: req.body.answers,
+         type: req.body.type
+      }
+      const question = await save(model);
+
+      res.status(200).send({payload: question})
+   } catch(error) { next(new AppError(error.message ))};
```
Comment (Gruseckis) [inline controllers/questionController.js]: Please style it properly content of catch block should be on new line

File: controllers/questionController.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
+import {
```
Comment (Gruseckis) [inline controllers/questionController.js]: This file should not be part of this pull request - only model

File: models/questionModel.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
+import mongoose from 'mongoose';
+
+const questionSchema = new mongoose.Schema(
+   {
+      "question": { type: String, required: true },
+      "correct": [ {type: String, required: true } ], // array of correct answer ID's 
+      "answers": [ {type: String, required: true } ], // array of possible answers as strings
+      "type": { enum: ['input', 'checkout', 'radio', 'textbox']}
```
Comment (Gruseckis) [inline models/questionModel.js]: You are missing type and required options in schema

File: models/questionModel.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+
+const QuestionModel = mongoose.model('Question', questionSchema);
+
+
+// save your question to DB
+const save = async model => new QuestionModel(model).save();
+
+
+// this will return all questions
+const getAllQuestions = async () => QuestionModel.find({});
```
Comment (Gruseckis) [inline models/questionModel.js]: You can just do find() no need to pass empty object there

File: models/questionModel.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+// get specific question by providing question ID
+const getQuestionByID = async _id => QuestionModel.findById({_id});
```
Comment (Gruseckis) [inline models/questionModel.js]: You can just pass id in this function no need to cast it to object

File: models/questionModel.js [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+// returns all questions based on provided array of questions ID's
+//const getQuestionsFromIdArray = async IDsArr => QuestionModel.find({ $in: IDsArr });
+
+// updates question by provided ID
+const updateQuestionByID = async (_id, update) => QuestionModel.findOneAndUpdate(_id, update, {new: true });
```
Comment (Gruseckis) [inline models/questionModel.js]: This should be findByIdAndUpdate

File: models/questionModel.js [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+
+// delete question by provided ID(only quis owner or admin can delete questions)
+const deleteQuestionByID = async _id => QuestionModel.findByIdAndDelete(_id);
+
+
+export  {
```
Comment (Gruseckis) [inline models/questionModel.js]: Schema and model should be exported aswell

File: routes/questionRoutes.js [hunk 10]
```diff
@@ -0,0 +1,23 @@
+import express from 'express';
```
Comment (Gruseckis) [inline routes/questionRoutes.js]: This should not be a part of this pull request. Please remove.

File: server.js [hunk 11]
```diff
@@ -5,6 +5,8 @@ import defaultErrorHandler from './middlewares/defaultErrorHandler';
 import mongoose from 'mongoose';
```
Comment (Gruseckis) [inline server.js]: These changes should not be part of this pull request

**Review by Gruseckis — CHANGES_REQUESTED** [review]

File: .env.template (hunk 1 above)
Comment (ghost) [reply .env.template]: done

File: controllers/questionController.js (hunk 2 above)
Comment (ghost) [reply controllers/questionController.js]: added const and exports at the bottom

File: controllers/questionController.js (hunk 3 above)
Comment (ghost) [reply controllers/questionController.js]: done

File: models/questionModel.js (hunk 5 above)
Comment (ghost) [reply models/questionModel.js]: added

File: models/questionModel.js [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+      "type":  { 
+                  type: String,
+                  required: true, 
+                  enum: ['input', 'checkout', 'radio', 'textbox']
+               }
+   },
+   { timestamps: true },
+);
+
+export const QuestionModel = mongoose.model('Question', questionSchema);
```
Comment (Gruseckis) [inline models/questionModel.js]: You should only have one export

File: models/questionModel.js [hunk 13]
```diff
@@ -0,0 +1,46 @@
+import mongoose from 'mongoose';
+
+export const questionSchema = new mongoose.Schema(
```
Comment (Gruseckis) [inline models/questionModel.js]: There should be only one export per file

File: server.js [hunk 14]
```diff
@@ -6,6 +6,8 @@ import mongoose from 'mongoose';
 import bodyParser from 'body-parser';
 import index from './routes/index';
 
+
```
Comment (Gruseckis) [inline server.js]: please remove extra white spaces

File: models/questionModel.js [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+
+
+// this will return all questions
+const getAllQuestions = async () => QuestionModel.find();
+
+// get specific question by providing question ID
+const getQuestionByID = async _id => QuestionModel.findById(_id);
+
+// returns all questions based on provided array of questions ID's
+//const getQuestionsFromIdArray = async IDsArr => QuestionModel.find({ $in: IDsArr });
```
Comment (Gruseckis) [inline models/questionModel.js]: If you are not using this you should remove it

**Review by Gruseckis — CHANGES_REQUESTED** [review]

**Review by Gruseckis — APPROVED** [review]
